api/mfx_dispatch/linux/device_ids.h

```diff
@@ -0,0 +1,494 @@
+    { 0x4698, MFX_HW_ADL_S, GT_INTEGRATED },//ADL-S
+    { 0x4699, MFX_HW_ADL_S, GT_INTEGRATED },//ADL-S
+};
+
+static inline const char *get_name(eMFXHWType platform) {
+    switch (platform) {
+    case MFX_HW_SNB: return "SNB";
+    case MFX_HW_IVB: return "IVB";
+    case MFX_HW_HSW: return "HSW";
+    case MFX_HW_HSW_ULT: return "HSW_ULT";
+    case MFX_HW_VLV: return "VLV";
+    case MFX_HW_BDW: return "BDW";
+    case MFX_HW_CHT: return "CHT";
+    case MFX_HW_SKL: return "SKL";
+    case MFX_HW_APL: return "APL";
+    case MFX_HW_KBL: return "KBL";
+    case MFX_HW_GLK: return "GLK";
+    case MFX_HW_CFL: return "CFL";
+    case MFX_HW_CNL: return "CNL";
+    case MFX_HW_ICL: return "ICL";
+    case MFX_HW_ICL_LP: return "ICL_LP";
+    case MFX_HW_JSL: return "JSL";
+    case MFX_HW_EHL: return "EHL";
+    case MFX_HW_TGL_LP: return "TGL_LP";
+    case MFX_HW_DG1: return "DG1";
+    case MFX_HW_RKL: return "RKL";
+    case MFX_HW_ADL_S: return "ADL_S";
+    case MFX_HW_UNKNOWN: default: return "UNKNOWN";
+    }
+}
+
+typedef struct {
+    int vendor_id;
+    int device_id;
+    eMFXHWType platform;
+    GraphicsType type;
+} Device;
+
+static inline eMFXHWType get_platform(int device_id) {
+    for (unsigned i = 0; i < sizeof(msdkDevIDs) / sizeof(msdkDevIDs[0]); ++i) {
+        if (msdkDevIDs[i].device_id == device_id) {
+            return msdkDevIDs[i].platform;
+        }
+    }
+    return MFX_HW_UNKNOWN;
+}
+
+static inline GraphicsType get_graphics_type(int device_id) {
+    for (unsigned i = 0; i < sizeof(msdkDevIDs) / sizeof(msdkDevIDs[0]); ++i) {
+        if (msdkDevIDs[i].device_id == device_id) {
+            return msdkDevIDs[i].type;
+        }
+    }
+    return GT_INTEGRATED;
+}
+
+std::vector <Device> get_devices() {
+    const char *dir = "/sys/class/drm";
+    const char *device_id_file = "/device/device";
+    const char *vendor_id_file = "/device/vendor";
+    int i = 0;
+    std::vector <Device> result;
+    for (; i < 64; ++i) {
+        Device device;
+        std::string path = std::string(dir) + "/renderD" + std::to_string(128 + i) + vendor_id_file;
+        FILE *file = fopen(path.c_str(), "r");
+        if (!file) break;
+        if (fscanf(file, "%x", &device.vendor_id) == EOF) break;
+        fclose(file);
+        if (device.vendor_id != 0x8086) {  // Filter out non-Intel devices
+            continue;
+        }
+        path = std::string(dir) + "/renderD" + std::to_string(128 + i) + device_id_file;
+        file = fopen(path.c_str(), "r");
+        if (!file) break;
+        if (fscanf(file, "%x", &device.device_id) == EOF) break;
+        fclose(file);
+        device.platform = get_platform(device.device_id);
+        device.type = get_graphics_type(device.device_id);
+        result.emplace_back(device);
+    }
+    std::sort(result.begin(), result.end(), [](const Device &a, const Device &b) {
+        if (a.type == b.type)
+            return a.device_id < b.device_id;
```

Comment:
Need to fix sorting.... It's not guaranteed, I think, that new devices will have devids > than existing ones. For example, see APL=0x0a84 vs. HSW=0x0D1A.

For what we care here in dispatcher, we can probably just sort per `eMFXHWType platform`. The only thing we actually need from such sort is: do we have new platforms on the system or not?

Reply:
Done sorting per platform